### DISCUSSION TITLE
Robo Tidy configuration file

### DISCUSSION THREAD

**Discussion by Syman123** [MarketSquare/robotframework-tidy#449] [Q&A]
Can someone support me with a fully configured configuration file with all transformers? Would it be possible to include it in the repo?

**Comment by bhirsz** [comment]
I was actually thinking about the same some time ago - to be able to generate a configuration file from what currently robotidy is currently using. We don't have this feature yet but I can provide you a file that's more or less close to robotidy default configuration (if you just run "robotidy ."):


Since those are defaults, you don't need to define them in the configuration - but it may be easier to tinker with the configuration having a "full" file. Please let me know if there are any issues with this file.

```
[tool.robotidy]
src = ["."]
exclude = "/(\.direnv|\.eggs|\.git|\.hg|\.nox|\.tox|\.venv|venv|\.svn)/"
skip_gitignore = false
overwrite = true
diff = false
color = true
check = false
spacecount = 4
indent = 4
continuation_indent = 4
lineseparator = "native"
separator = "space"
line_length = 120
language = "en"
configure = [
    "AddMissingEnd:enabled=True",
    "AlignKeywordsSection:enabled=False",
    "AlignSettingsSection:enabled=True",
    "AlignTemplatedTestCases:enabled=False",
    "AlignTestCasesSection:enabled=False",
    "AlignVariablesSection:enabled=True",
    "DiscardEmptySections:enabled=True",
    "IndentNestedKeywords:enabled=False",
    "InlineIf:enabled=True",
    "MergeAndOrderSections:enabled=True",
    "NormalizeAssignments:enabled=True",
    "NormalizeNewLines:enabled=True",
    "NormalizeSectionHeaderName:enabled=True",
    "NormalizeSeparators:enabled=True",
    "NormalizeSettingName:enabled=True",
    "NormalizeTags:enabled=True",
    "OrderSettings:enabled=True",
    "OrderSettingsSection:enabled=True",
    "OrderTags:enabled=False",
    "RemoveEmptySettings:enabled=True",
    "RenameKeywords:enabled=False",
    "RenameTestCases:enabled=False",
    "ReplaceBreakContinue:enabled=True",
    "ReplaceReturns:enabled=True",
    "ReplaceRunKeywordIf:enabled=True",
    "SmartSortKeywords:enabled=False",
    "SplitTooLongLine:enabled=True",
    "Translate:enabled=False",
]
```

This does not include the default configuration of the transformers. If you need it - let me know. For example for the SplitTooLongLine transformer the full default configuration would be:
```
configure = [
    "SplitTooLongLine:enabled=True",
    "SplitTooLongLine:line_length=120",  # line_length is inherited from the global --line-length by default, but it can be overwritten just for this transformer
    "SplitTooLongLine:split_on_every_arg=True",
    "SplitTooLongLine:split_on_every_value=True"
]
```

**Reply by bhirsz** [reply]
There are also some settings that enables some features of the robotidy (starting from robotidy --list which list the transformers, or robotidy --skip-x that skip formatting x type - this is described in our docs).
